tensorflow/lite/kernels/concatenation.cc

```diff
@@ -54,7 +48,6 @@ TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
 
   // TODO(ahentz): These are limitations of our implementation that could be
   // removed with a bit of effort.
-  TF_LITE_ENSURE(context, t0->dims->size <= 4);
```

Comment:
Please avoid coupling refactorings with behavioral changes. This should be its own CL, with its own test, separate from any kind of refactoring.